allensdk/brain_observatory/behavior/session_apis/data_transforms/behavior_ophys_data_xforms.py

```diff
@@ -206,13 +208,33 @@ def get_cell_roi_ids(self):
     def get_raw_dff_data(self):
         dff_path = self.get_dff_file()
         with h5py.File(dff_path, 'r') as raw_file:
-            dff_traces = np.asarray(raw_file['data'])
+            raw_dff_traces = np.asarray(raw_file['data'])
+            roi_names = np.asarray(raw_file['roi_names']).astype(int)
+
+        # guarantee that DFF traces are ordered the same
+        # way as ROIs in the cell_specimen_table
+        cell_roi_id_list = self.get_cell_roi_ids().astype(int)
+
+        if not np.in1d(roi_names, cell_roi_id_list).all():
+            raise RuntimeError("DFF traces contains ROI IDs that "
+                               "are not in cell_specimen_table.cell_roi_id")
+        if not np.in1d(cell_roi_id_list, roi_names).all():
+            raise RuntimeError("cell_specimen_table contains ROI IDs "
+                               "that are not in DFF traces file")
+
+        dff_traces = np.zeros(raw_dff_traces.shape, dtype=float)
+        for ii in range(dff_traces.shape[0]):
```

Comment:
this might be a little more clear:
```
for raw_trace, name in zip(raw_dff_traces, roi_names):
    idx = np.where(cell_roi_id_list==name)[0][0]
    dff_traces[idx, :] = trace
```

Reply:
Good idea. Implemented.